ocs_ci/framework/pytest_customization/ocscilib.py

```diff
@@ -622,14 +622,24 @@ def pytest_runtest_makereport(item, call):
     # we only look at actual failing test calls, not setup/teardown
     if rep.failed and ocsci_config.RUN.get("cli_params").get("collect-logs"):
         test_case_name = item.name
-        ocp_logs_collection = True if rep.when == "call" else False
-        mcg = (
+        ocp_logs_collection = (
```

Comment:
Do we cover the scenario that we are collecting OCP must gather logs at least once at the end of the execution? With that collect logs on success?

Reply:
I believe this is handled separately in a session scope fixture finalizer here: https://github.com/red-hat-storage/ocs-ci/blob/master/tests/conftest.py#L3710-L3726

Reply:
However, with the change to the default value for `ocp` below in `utils.py` these calls will need to be updated as it will no longer work as intended. We are relying on the defaults to be True for those calls. @OdedViner

Reply:
done